p2p/party_coordinator_test.go

```diff
@@ -103,23 +163,83 @@ func TestPartyCoordinatorTimeOut(t *testing.T) {
 
 	msgID := conversion.RandStringBytesMask(64)
 	wg := sync.WaitGroup{}
-	expected := peers[:2]
-	sort.Strings(expected)
+	leader, err := LeaderNode(msgID, 10, peers)
+	assert.Nil(t, err)
+
+	// we sort the slice to ensure the leader is the first one easy for testing
+	for i, el := range pcs {
+		if el.host.ID().String() == leader {
+			if i == 0 {
+				break
+			}
+			temp := pcs[0]
+			pcs[0] = el
+			pcs[i] = temp
+			break
+		}
+	}
+	assert.Equal(t, pcs[0].host.ID().String(), leader)
+
+	// we test the leader is offline
+	for _, el := range pcs[1:] {
+		wg.Add(1)
+		go func(coordinator *PartyCoordinator) {
+			defer wg.Done()
+			sigChan := make(chan string)
+			_, _, err := coordinator.JoinPartyWithLeader(msgID, 10, peers, 3, sigChan)
+			assert.Equal(t, err, ErrLeaderNotReady)
+		}(el)
 
-	for _, el := range pcs[:2] {
+	}
+	wg.Wait()
+	// we test one of node is not ready
+	var expected []string
+	for _, el := range pcs[:3] {
+		expected = append(expected, el.host.ID().String())
+		sort.Strings(expected)
 		wg.Add(1)
 		go func(coordinator *PartyCoordinator) {
 			defer wg.Done()
-			onlinePeers, err := coordinator.JoinPartyWithRetry(msgID, peers)
-			assert.Errorf(t, err, ErrJoinPartyTimeout.Error())
+			sigChan := make(chan string)
+			onlinePeers, _, err := coordinator.JoinPartyWithLeader(msgID, 10, peers, 3, sigChan)
+			assert.Equal(t, ErrJoinPartyTimeout, err)
 			var onlinePeersStr []string
 			for _, el := range onlinePeers {
 				onlinePeersStr = append(onlinePeersStr, el.String())
 			}
 			sort.Strings(onlinePeersStr)
-			assert.EqualValues(t, onlinePeersStr, expected)
+			sort.Strings(expected[:3])
+			assert.EqualValues(t, expected, onlinePeersStr)
 		}(el)
 	}
-
 	wg.Wait()
 }
+
+func TestGetPeerIDs(t *testing.T) {
+	id1 := tnet.RandIdentityOrFatal(t)
+	mn := mocknet.New()
+	// add peers to mock net
+
+	a1 := tnet.RandLocalTCPAddress()
+	h1, err := mn.AddPeer(id1.PrivateKey(), a1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	p1 := h1.ID()
+	timeout := time.Second * 2
+	pc := NewPartyCoordinator(h1, timeout, zerolog.Nop())
+	r, err := pc.getPeerIDs([]string{})
+	assert.Nil(t, err)
+	assert.Len(t, r, 0)
+	input := []string{
+		p1.String(),
+	}
+	r1, err := pc.getPeerIDs(input)
+	assert.Nil(t, err)
+	assert.Len(t, r1, 1)
+	assert.Equal(t, r1[0], p1)
+	input = append(input, "whatever")
+	r2, err := pc.getPeerIDs(input)
+	assert.NotNil(t, err)
+	assert.Len(t, r2, 0)
```

Comment:
I don't think this one necessary, other values should be ignored if error is returned